inspire_matcher/api.py

```diff
@@ -62,6 +63,13 @@ def match(record, config=None):
         raise KeyError('Malformed configuration: %s.' % repr(e))
 
     source = config.get('source', [])
+    match_deleted = config.get('match_deleted', False)
+    collections = config.get('collections')
+    if not (collections is None or (
+            isinstance(collections, (list, tuple)) and
+            all(isinstance(collection, string_types) for collection in collections)
+    )):
```

Comment:
not sure about this, on the one hand this is not done for other flags, but I feel a bit uneasy about not having any checks that the config is sound.

Reply:
can we make it a bit more readable?

Reply:
not easily, as we short circuiting for it to work correctly.